### PR TITLE
upgrade to kernel 5.3.1

### PR DESCRIPTION
**Pull request by rk-zero**
i have hacked together all needed changes to go to 5.3.1 and have HANTRO extensions enabled

You may review the changes to the buildKernel.sh as i have just commented out 2 lines of 5.2.x patches

i removed 2 patches but left other files intact. you may simply apply another commit to remove those if wanted.

makefile is fixed to remove downloaded linux-libre tar, so FRESH will get triggered.

boots fine, panfrost mesa works after compilation, no hangs on resume, etc.

Please also review the kernel config, as i enabled CONFIG_NET_VENDOR_GOOGLE because the commit said something about better lid angle detection - maybe not needed on c201p.

### CODE REVIEW

**Comment by rk-zero** [comment]
CompilePanfrost.sh under scripts/InstallResources is not copued on the image, maybe someone can pinpoint me in the right direction why that is - the buildFilesystem script just cp the whole directory, still it's missing.

**Comment by gdallasdye** [comment]
I can confirm that this works for me to. Previously I had a similar script but missed the build-dep mesa line and was installing enough dependencies by hand. Weston worked, gnome 3 wayland worked for a couple of reboots, gnome 3 xorg had glitchy artifacts. On the other hand your script has better and more reproducible results and better xorg support. Tested on both 5.2 and 5.3. 

Works:
- Watching video in firefox at 60fps. (Gnome3, Wayland) (FF7 Remake trailer used for reference)
- Kodi watching live tv (mpeg2 480i and h264 720p). Kodi zoom and deinterlace options work too. News ticker at the bottom of *** News scrolls smoothly. Terminal shows: MESA: User error: GL_INVALID_ENUM in glDisable(GL_TEXTURE_2D). This error is not noticeable without terminal. 
- Gnome shell animations (wayland)
- HDMI video. Need to do more performance testing.

Doesn't work:
- Chromium video playback (xorg and wayland) at acceptable quality (and epiphany, midori, eolie, presumably other webkit-based). Can't post output from terminal now, but as I remember it's cannot create DRI screen, failed to load rockchip driver, failed to load swrast, gl errors (Gnome3, Xorg, Wayland)
- HDMI audio

What crashes:

- Synaptic (wayland and xorg) crashes the entire session; xhost +local: && sudo synaptic
- Gnome settings (xorg) (entire session)

Finishing notes: 

You compile script probably didn't make it in the image because there was already a base image generated. It appeared in mine.

**Comment by SolidHal** [comment]
@rk-zero the kernel patches are applied if the extracted linux source isn't found, so the linux-libre-KVER-gnu.tar.lz doesn't need  to be removed. 

```
# build Linux-libre, with ath9k_htc
[ ! -f linux-libre-$KVER-gnu.tar.lz ] && wget https://www.linux-libre.fsfla.org/pub/linux-libre/releases/$KVER-gnu/linux-libre-$KVER-gnu.tar.lz
[ ! -d linux-$KVER ] && tar --lzip -xvf linux-libre-$KVER-gnu.tar.lz && FRESH=true
cd linux-$KVER
make clean
make mrproper
#Apply the usb and mmc patches if unapplied
[ "$FRESH" = true ] && for i in $RESOURCES/patches-tested/kernel/5.x-dwc2/*.patch; do echo $i; patch -p1 < $i; done
[ "$FRESH" = true ] && for i in $RESOURCES/patches-tested/DTS/5.x-dts/*.patch; do echo $i; patch -p1 < $i; done
[ "$FRESH" = true ] && for i in $RESOURCES/patches-tested/DTS/*.patch; do echo $i; patch -p1 < $i; done
[ "$FRESH" = true ] && for i in $RESOURCES/patches-tested/kernel/*.patch; do echo $i; patch -p1 < $i; done

```

**Comment by SolidHal** [comment]
@rk-zero thanks for the PR!

> Please also review the kernel config, as i enabled CONFIG_NET_VENDOR_GOOGLE because the commit said something about better lid angle detection - maybe not needed on c201p.

Could you direct me to the commit that says that? All the config option seems to do is enable support for google ethernet devices, which aren't present on the c201

File: scripts/InstallScripts/InstallPackages.sh [hunk 1]
```diff
@@ -57,7 +57,7 @@ then
   # is told to sleep at lid close, and activate lock
   # gnome-screensaver shows the desktop for a fraction of a second at wakeup
   # xscreensaver works as well, if you prefer that but is less simple
-  apt install -y -t unstable xsecurelock
+  DEBIAN_FRONTEND=noninteractive apt install -y -t unstable xsecurelock
```
Comment (SolidHal) [inline scripts/InstallScripts/InstallPackages.sh]: Funny enough, I've had this hanging around on my build for a bit. Seems to work in my experience.

**Comment by SolidHal** [comment]
> CompilePanfrost.sh under scripts/InstallResources is not copied on the image, maybe someone can pinpoint me in the right direction why that is - the buildFilesystem script just cp the whole directory, still it's missing.

Are you making a new "BASE" filesystem between tests? If the "Prawnos-blah-blah.img-BASE" exists a new filesystem won't be created, so your script won't get included.

**Comment by SolidHal** [comment]
I really appreciate your mesa script which I would  like to include so more people can test out panfrost, and the tiling keybindings which have been on my list of todos for awhile. Thanks for addressing both of those. 

I noticed some issues in the kernel config, and I don't think all of the 5.x patches made it into 5.3.1 so I can't take this PR as is. 

I'll be working on getting the kernel upgraded for better panfrost support this week. If you'd like, you can make PRs for the mesa build script and the keybinding changes or I can cherry-pick them. No biggie either way. 
Thanks again for all of your work!

**Comment by rk-zero** [comment]
Sorry, i mixed that up, i meant;

CONFIG_IIO_CROS_EC_SENSORS_LID_ANGLE=y

Is said to detect the lid angle on some chromeos devices, https://github.com/torvalds/linux/tree/master/drivers/iio/common/cros_ec_sensors

Ves, it seems CONFIG_NET_VENDOR_GOOGLE is indeed not needed, maybe one could CONFIG_GOOGLE_FIRMWARE too.

**Comment by rk-zero** [comment]
This is my first work in github and kernel stuff, i don't mind feedback instead of a merge at all!

I too have issues with 5.3.1 still and am totally fine with playing with my own fork to test.

Thanks so much for the feedback!

**Comment by rk-zero** [comment]
Hey @gdallasdye and thanks for your report.

> Works:
> 
>     * Watching video in firefox at 60fps. (Gnome3, Wayland) (FF7 Remake trailer used for reference)

Firefox sometimes has no sound in youtube videos, but without compositors enabled video is fine. may use h264ify.

> 
>     * Kodi watching live tv (mpeg2 480i and h264 720p). Kodi zoom and deinterlace options work too. News ticker at the bottom of *** News scrolls smoothly. Terminal shows: MESA: User error: GL_INVALID_ENUM in glDisable(GL_TEXTURE_2D). This error is not noticeable without terminal.
> 
>     * Gnome shell animations (wayland)
> 

I would like to introduce sway into PrawnOS, because there are still so much issues with compositors enabled. I will try a crashless 5.2 kernel with panfrost if you have positive experience with wayland so far.

>     * HDMI video. Need to do more performance testing.



> Doesn't work:
> 
>     * Chromium video playback (xorg and wayland) at acceptable quality (and epiphany, midori, eolie, presumably other webkit-based). Can't post output from terminal now, but as I remember it's cannot create DRI screen, failed to load rockchip driver, failed to load swrast, gl errors (Gnome3, Xorg, Wayland)

I switched to 5.3.1 for hantro support, there is work to enable playback in chromium: https://chromium.googlesource.com/chromiumos/third_party/libv4lplugins/+/refs/heads/master
> 
>     * HDMI audio
> 
> 
> What crashes:
> 
>     * Synaptic (wayland and xorg) crashes the entire session; xhost +local: && sudo synaptic
> 
>     * Gnome settings (xorg) (entire session)

Sometimes x crashed and throughs me back to the login screen, mostly when opening gimp and resizing windows or doing more stressful work on the gpu. Am to stupid to debug that for now.

Sometimes machines left alone lock up completely, also to noob to catch errors then, any help is appreciated.

**Comment by rk-zero** [comment]
I will close this PR as there are now too many differences between branches.

**Comment by SolidHal** [comment]
> This is my first work in github and kernel stuff, i don't mind feedback instead of a merge at all!
> 
> I too have issues with 5.3.1 still and am totally fine with playing with my own fork to test.
> 
> Thanks so much for the feedback!

I appreciate all of your work, and hope to see future PRs!

**Pull request closed without merging**